src/InferenceFramework.cpp

```diff
@@ -38,15 +38,17 @@ int InferenceFramework::GetTimeOffSet(const ModelInputParameters& modelInputPara
 
 void InferenceFramework::Run()
 {
-    	/*---------------------------------------
+    const unsigned int nInferenceParams = Model::ModelParameters::NPARAMS;
```

Comment:
This is a valid solution for the sake of code readability but given that `modelInputParameters_` still exists in this function and has `nPar` as a member and given that `nPar` is read from "parameters.ini", it might be better to use `modelInputParameters_.nPar`.

Reply:
Broadly agree, but this is actually tied to a bigger problem, which is that I don't think that `nPar` should be in the `parameters.ini` at all, as it is not possible in general to vary it without breaking the code. I'll bring it up with Thibaud and see what he thinks about removing it.